array.c

```diff
@@ -4291,6 +4291,71 @@ rb_ary_or(VALUE ary1, VALUE ary2)
     return ary3;
 }
 
+
+/*
+ *  call-seq:
+ *     ary.union(other_ary1, other_ary2,...)   -> ary
+ *
+ *  Set Union --- Returns a new array by joining +other_ary+s with +self+,
+ *  excluding any duplicates and preserving the order from the given arrays.
+ *
+ *  It compares elements using their #hash and #eql? methods for efficiency.
+ *
+ *     [ "a", "b", "c" ].union( [ "c", "d", "a" ] )    #=> [ "a", "b", "c", "d" ]
+ *     [ "a" ].union( [ ["e", "b"], ["a", "c", "b"] ] )    #=> [ "a", "e", "b", "c" ]
+ *     [ "a" ].union #=> [ "a" ]
+ *
+ *  See also Array#|.
+ */
+
+static VALUE
+rb_ary_union_multi(int argc, VALUE *argv, VALUE ary)
+{
+    int i, sum;
+    VALUE hash, ary_union;
+
+    sum = RARRAY_LEN(ary);
+    for (i = 0; i < argc; i++) sum += RARRAY_LEN(argv[i]);
```

Comment:
Non array argument causes a segfault.

Reply:
Please try: `[].union "garbage", :junk, 5, nil`

Ruby will crash because you didn't check if `argv[i]` is an Array type before calling `RARRAY_LEN()`, you check the type (maybe [`rb_check_array_type()`](https://github.com/ruby/ruby/blob/6b818dd9616ce6f0c849ab32c5869cc55b624b0a/array.c#L652)) first